### PR TITLE
tbb: install pkg-config file

### PR DESCRIPTION
**Pull request by jtojnar**
###### Motivation for this change
Stable versions of tbb (a gazebo dependency) do not provide a pkg-config file. And while gazebo’s build system can still find the library without .pc file and successfully link against it, it will not propagate the necessary flags in the .pc file. This will result in libraries linking against gazebo not being able to find tbb header:

    In file included from /nix/store/big05g101j4angkwpxv9n0xhml8j6bgn-gazebo-11.3.0/include/gazebo-11/gazebo/Master.hh:28,
                     from /home/jtojnar/Projects/RoFI/softwareComponents/simplesim/gazebo_master/gazebo_master.hpp:6,
                     from /home/jtojnar/Projects/RoFI/softwareComponents/simplesim/gazebo_master/gazebo_master.cpp:1:
    /nix/store/big05g101j4angkwpxv9n0xhml8j6bgn-gazebo-11.3.0/include/gazebo-11/gazebo/transport/Connection.hh:20:10: fatal error: tbb/task.h: No such file or directory
       20 | #include <tbb/task.h>
          |          ^~~~~~~~~~~~

This issue does not occur on Debian because it [ships custom `tbb.pc` file][1] so `gazebo.pc` will [include `tbb` in `Requires` field][2] there.

This should be fixed in an upcoming tbb release since it will have an [upstream pkg-config file][3]. But we will still need to fix it upstream as per https://github.com/jtojnar/cmake-snips#concatenating-paths-when-building-pkg-config-files.

[1]: https://salsa.debian.org/science-team/tbb/-/blob/4c31cbb2fa413f099cb4207beb10757dc2f41674/debian/tbb.pc.in
[2]: https://github.com/osrf/gazebo/commit/417514735dbc03d6357bd57e9f2f59d4198544a2#diff-070f9711d42593f839e9cb29c2bcb31f999939af11125c208965891a2ba1617c
[3]: https://github.com/oneapi-src/oneTBB/tree/6aa706d879bebf3873a4b94a4b1a27aac4f844e8/integration/pkg-config

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/tbb/default.nix [hunk 1]
```diff
@@ -1,22 +1,46 @@
...
-    sha256 = "sha256-prO2O5hd+Wz5iA0vfrqmyHFr0Ptzk64so5KpSpvuKmU=";
+    owner = "oneapi-src";
+    repo = "oneTBB";
+    rev = "v${version}";
+    sha256 = "prO2O5hd+Wz5iA0vfrqmyHFr0Ptzk64so5KpSpvuKmU=";
   };
 
-  nativeBuildInputs = optional stdenv.isDarwin fixDarwinDylibNames;
+  patches = [
+    (fetchurl {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/tbb/default.nix:22]: Please use fetchpatch.

File: pkgs/development/libraries/tbb/default.nix [hunk 2]
```diff
@@ -1,22 +1,46 @@
...
+      url = "https://github.com/openembedded/meta-openembedded/raw/39185eb1d1615e919e3ae14ae63b8ed7d3e5d83f/meta-oe/recipes-support/tbb/tbb/GLIBC-PREREQ-is-not-defined-on-musl.patch";
+      sha256 = "gUfXQ9OZQ82qD6brgauBCsKdjLvyHafMc18B+KxZoYs=";
+    })
 
-  makeFlags = optional (compiler != null) "compiler=${compiler}"
-    ++ optional (stdver != null) "stdver=${stdver}";
+    (fetchurl {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/tbb/default.nix:27]: Please use fetchpatch.

File: pkgs/development/libraries/tbb/default.nix [hunk 3]
```diff
@@ -1,22 +1,46 @@
-{ lib, stdenv, fetchFromGitHub, fixDarwinDylibNames, compiler ? if stdenv.cc.isClang then "clang" else null, stdver ? null }:
+{ lib
+, stdenv
+, fetchurl
+, fetchFromGitHub
+, fixDarwinDylibNames
+, compiler ? if stdenv.cc.isClang then "clang" else null
+, stdver ? null
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/tbb/default.nix:7]: Do these need to be overwritable?

Comment (jtojnar) [reply pkgs/development/libraries/tbb/default.nix:7]: Who knows :woman_shrugging:

**Pull request merged**